crates/task-impls/src/consensus.rs

```diff
@@ -403,20 +401,27 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                             warn!("Timeout certificate for view {} was invalid", *view);
                             return;
                         }
-                    } else if let Some(cert) = &self.view_sync_cert {
-                        // View sync certs _must_ be for the current view.
-                        if cert.view_number != view {
-                            debug!(
+                    } else if let Some(cert) = &self.proposal_cert {
+                        match cert {
+                            either::Left(_) => {
+                                error!("Timeout cert found during proposal recv");
+                            }
+                            either::Right(view_sync) => {
+                                // View sync view_syncs _must_ be for the current view.
+                                if view_sync.view_number != view {
+                                    debug!(
                                 "Cert view number {:?} does not match proposal view number {:?}",
-                                cert.view_number, view
+                                view_sync.view_number, view
                             );
```

Comment:
Not at all a strong opinion, but just a thought I had while reading this -- would it be better to use a custom enum? Not sure what it should be called but e.g. `pub enum ViewChangeEvidence = { Timeout(..), ViewSync(..) }`.

My worry is: if I mix up `Left` and `Right` here, I don't think I'd get a type-error because both certificates have `view_number`. And I think it would be easy to miss in code review, because the names `Left` and `Right` aren't descriptive.

Reply:
Following up: After discussion with the team we have all decided that we prefer Salman's idea. Rolling this into my next commit.